### PR TITLE
MAINT: proper random number initialization and readability fix

### PR DESCRIPTION
**Pull request by pmla**
#### What does this implement/fix?

This fixes a couple of issues that I overlooked in #10776.
* Seed for generation of random rotations is properly initialized.
* Variable re-use makes the code more readable

#### Additional information

This PR is intended to make #10692 more digestible.

### CODE REVIEW

File: scipy/spatial/tests/test_spherical_voronoi.py [hunk 1]
```diff
@@ -183,8 +183,7 @@ def test_rank_deficient(self):
     @pytest.mark.parametrize("radius", [0.5, 1, 2])
     @pytest.mark.parametrize("center", [(0, 0, 0), (1, 2, 3)])
     def test_geodesic_input(self, n, radius, center):
-        np.random.seed(0)
-        U = Rotation.random().as_matrix()
+        U = Rotation.random(random_state=0).as_matrix()
```
Comment (tylerjereddy) [inline scipy/spatial/tests/test_spherical_voronoi.py]: @nmayorov Can you confirm this is an improvement? I have a hard time keeping track of the changes in `random` machinery in NumPy & what is going on with `Rotation` class.

**Review by tylerjereddy — COMMENTED** [review]
Thanks Peter, this should be good---I just asked Nikolay to check one thing I'm not sure about.

**Comment by tylerjereddy** [comment]
`ppc64le` failure look unrelated--I think this is a different issue from one @peterbell10 saw recently in another PR..

File: scipy/spatial/tests/test_spherical_voronoi.py (hunk 1 above)
Comment (nmayorov) [reply scipy/spatial/tests/test_spherical_voronoi.py]: I think this is a usual approach to ensure test repeatability.

Nothing special here, looks reasonable to me.

Comment (tylerjereddy) [reply scipy/spatial/tests/test_spherical_voronoi.py]: I guess I'm not clear on the difference before vs. after. Anyway, if you aren't worried that's good.

**Comment by tylerjereddy** [comment]
In it goes, thanks Peter.

**Pull request merged**